### PR TITLE
Clean XAPI code by removing unused code and update to python3.

### PR DESCRIPTION
**Pull request by ashwin9390**
Main task CP-47869- Remove unused Xapi python2 scripts

subtask
CP-49681-Convert the existing code to python3 in Xapi

CP-49682-Fix pytype reporter error during python code test for python 3.11
-pytype_reporter: Expected to fail: scripts/hatests, scripts/ fatal: ambiguous argument 'scripts/hatests': unknown revision or path not in the working tree.

CP-49686-Fix ocaml test issue during static code analysis.
-When the initial removal files were given we had a requirement to remove scripts under scripts/template which lead to ocaml test make install  smoke test fail

XenRT test 1:

Test result -198930 (Dev Run)
Ring0 BST - 8.4.0-206415 (pbb-ashwinh-xapiclean) (repo) run 30 May 2024
Tests Executed (excludes blocked): 47
Tests Failed: 1
Tests Errored: 0
Tests Blocked: 0
Tests Not Executed: 0
Test failed due to production issues: 0
Pass rate (excluding failures due to non product issues): 100.0%
Pass rate (of testcases XenRT was able to resource): 97.9%
Pass rate (of all test cases): 97.9%
Percentage of nightly complete: 100.0%
Tests excluded from numbers: 0

XenRT test 2:

XenRT job 4014794 -pass

 PVSProxyBase/PVSProxyDeviceLifeCycleOps.win10-x64: pass
PVSProxyBase/PVSProxyServiceKill.win10-x64: pass
PVSProxyBase/PVSProxyServiceStop.win10-x64: pass
PVSProxyBase/PVSProxyOnRunningDevice.win10-x64: pass
PVSProxyBase/PVSProxyAfterHostReboot.win10-x64: pass
PVSProxyBase/PVSProxyChangeDom0Memory.win10-x64: pass
PVSProxyBase/PVSProxyDeviceLifeCycleOps.ws22-x64: pass
PVSProxyBase/PVSProxyServiceKill.ws22-x64: pass
PVSProxyBase/PVSProxyServiceStop.ws22-x64: pass
PVSProxyBase/PVSProxyOnRunningDevice.ws22-x64: pass
PVSProxyBase/PVSProxyAfterHostReboot.ws22-x64: pass
PVSProxyBase/PVSProxyChangeDom0Memory.ws22-x64: pass

Signed-off-by: Ashwinh <ashwin.h@cloud.com>

@stephenchengCloud @liulinC @psafont

### CODE REVIEW

**Comment by edwintorok** [comment]
I'd suggest to put the removals into a separate PR (they are already separate commits) , that should be easier to review and merge than the python updates.

**Comment by ashwin9390** [comment]
> I'd suggest to put the removals into a separate PR (they are already separate commits) , that should be easier to review and merge than the python updates.

-Sure, will do that .

**Comment by bernhardkaindl** [comment]
>> I'd suggest to put the removals into a separate PR (they are already separate commits) , that should be easier to review and merge than the python updates.
>
> -Sure, will do that .

I'd suggest putting the removals of tests and other scripts into separate PRs from removing examples from `scripts/examples`:

For example, this Ocaml task in this PR fails with an error

https://github.com/xapi-project/xen-api/actions/runs/9314471378/job/25638715050?pr=5662
```py
* Missing dependency:
- xen-api-sdk
  unknown package
```

If you split this PR into such smaller pieces, that avoids the need to debug now on which change causes it.

And you do need to do it all at once:

Just take it easy by working in PR after PR.

I'd suggest submitting changes together which thematically belong together:
1. Removal of examples (or fixing errors in them them)
   For example, you can likely remove examples like those:
   - `scripts/examples/smapiv2.py` - this example has 
been broken during the Python3 changes, with this change:
```
   -        log_f = open(log_file, "aw")
+        try:
+            log_f = io.open(log_file, "a")
+        except FilenotFoundError:
+            log_f = io.open(log_file, "w")
   ```
  -   (FilenotFoundError is not defined: it would have to be FileNotFoundError)
  ```py
   pytype: line 16, in reopenlog: Name 'FilenotFoundError' is not defined [name-error]
   ```
  - `scripts/examples/python/mini-xenrt.py` also had been broken during Python3 cannot work because `xmlrpc.client` has no `server`:
    
```py
    mini-xenrt.py", line 112, in <module>: No attribute 'server' on module 'xmlrpc.client' [module-attr]
   ```
    
2. Removals of other scripts that need to be looked at closer, like the removal of message-switch
[ocaml/message-switch/python/message_switch.py](https://github.com/xapi-project/xen-api/pull/5662/files#diff-aa08b8ecc6fa6b703fc033a8e65361245b1dcdd7e3ba9588a484dc6348a8f45e) - I thought it is used?
3. Removals of tests
4. Pytnon2 to Python3 migrations.

File: ocaml/xcp-rrdd/scripts/rrdd/rrdd-example.py [hunk 1]
```diff
@@ -1,18 +0,0 @@
-#!/usr/bin/env python
-
-import rrdd, os
-
-if __name__ == "__main__":
```
Comment (psafont) [inline ocaml/xcp-rrdd/scripts/rrdd/rrdd-example.py:5]: Is the anything that documents the right usage? From the GFS2 component to report, we know it's hard to do it right, so having an example is useful. maybe in the developer docs, like https://xapi-project.github.io/devguide/howtos/metrics.html

Comment (ashwin9390) [reply ocaml/xcp-rrdd/scripts/rrdd/rrdd-example.py:5]: The link is broken , found an initial commit ,nothing more about the usage.

commit 7522520faa15d8ae98d0530237a61598289d2aca
Author: Rok Strniša <rok.strnisa@citrix.com>
Date: Sat Jun 30 18:00:40 2012 +0100

Added python library rrdd.py, and example plugin rrdd-example.py.

Old XenServer(around 2014) had  rrdd-examples file under   /opt/xensource/libexec/ 

-rwxr-xr-x 1 root root      761 Jul 24 14:59 rrdd-example.py
-rw-r--r-- 1 root root      737 Jul 24 14:59 rrdd-example.pyc
-rw-r--r-- 1 root root      737 Jul 24 14:59 rrdd-example.pyo

Comment (psafont) [reply ocaml/xcp-rrdd/scripts/rrdd/rrdd-example.py:5]: The link is broken because it's an example of where it should be placed :)

**Pull request closed without merging**